### DISCUSSION TITLE
How would I use Blitz.js with the exisiting auth system?

### DISCUSSION THREAD

**Discussion by heauton** [blitz-js/blitz#2862] [Q&A] [answered]
I use Supabase Auth for one of my production projects. Is it possible to use Blitz.js with Supabase Auth? I looked through docs but authentication seemed to be built-in. In general, I couldn't find information on migrating from existing codebases. (Found https://github.com/blitz-js/blitz/discussions/2054, but no answers) - Any guidance will be helpful. Thanks!

**Comment by flybayer** [comment, marked as answer]
Good question! I looked through the Supabase docs and examples, and it seems they only support client-side auth. So to have authenticated server calls, you will need to use something in addition to supabase (like blitz auth).

What you can do, I think, is replace the default logic inside the blitz login mutation to instead call `supabase.auth.signIn()`. And if successful, call `ctx.session.$create()` like normal blitz auth.

So in essence, you are using supabase to manage the **identity verification**, but then using blitz to manage the **session**.

**Reply by heauton** [reply]
Ah, I see. Thanks! I'll give it a shot.

**Reply by flybayer** [reply]
Ok let us know how it goes!

**Reply by beerose** [reply]
Idea: `Using Blitz with custom auth` or `Using Blitz with Supabase` could be a nice addition to the docs.
